x/ibc/mock/bank/client/cli/tx.go

```diff
@@ -0,0 +1,128 @@
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	ics23 "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+	"github.com/cosmos/cosmos-sdk/x/ibc/mock/bank/internal/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func GetTxCmd(cdc *codec.Codec) *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:   "ibcmockbank",
+		Short: "IBC mockbank module transaction subcommands",
+		// RunE:  client.ValidateCmd,
+	}
+	txCmd.AddCommand(
+		GetTransferTxCmd(cdc),
+		GetMsgRecvPacketCmd(cdc),
+	)
+
+	return txCmd
+}
+
+func GetTransferTxCmd(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "transfer --src-port <src port> --src-channel <src channel> --denom <denomination> --amount <amount> --receiver <receiver> --source <source>",
+		Short: "Transfer tokens across chains through IBC",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			ctx := context.NewCLIContext().WithCodec(cdc).WithBroadcastMode(flags.BroadcastBlock)
+
+			sender := ctx.GetFromAddress()
+			receiver := viper.GetString(FlagReceiver)
+			denom := viper.GetString(FlagDenom)
+			srcPort := viper.GetString(FlagSrcPort)
+			srcChan := viper.GetString(FlagSrcChannel)
+			source := viper.GetBool(FlagSource)
+
+			amount, ok := sdk.NewIntFromString(viper.GetString(FlagAmount))
+			if !ok {
+				return fmt.Errorf("invalid amount")
+			}
+
+			msg := types.NewMsgTransfer(srcPort, srcChan, denom, amount, sender, receiver, source)
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return utils.GenerateOrBroadcastMsgs(ctx, txBldr, []sdk.Msg{msg})
+		},
+	}
+
+	cmd.Flags().AddFlagSet(FsTransfer)
+
+	cmd.MarkFlagRequired(FlagSrcPort)
+	cmd.MarkFlagRequired(FlagSrcChannel)
+	cmd.MarkFlagRequired(FlagDenom)
+	cmd.MarkFlagRequired(FlagAmount)
+	cmd.MarkFlagRequired(FlagReceiver)
+
+	cmd = client.PostCommands(cmd)[0]
+
+	return cmd
+}
+
+// GetMsgRecvPacketCmd returns the command to create a MsgRecvTransferPacket transaction
+func GetMsgRecvPacketCmd(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "recv-packet [/path/to/packet-data.json] [/path/to/proof.json] [height]",
```

Comment:
Is there a query that returns the data required for this command? Should be able to be constructed like:

```
recv-packet $(gaiacli q ibc packet-data) $(gaiacli q ibc channel packet-proof)
```

Also ideally the `proof` has the height returned with it so that would remove the need for the 3rd arg.

Reply:
It's seems that we should only store the packet hash. There is more than one proof that needs to be queried, should we add a common command to query the proof by the stored key or add different commands to query the connection/channel/packet_hash proof?